pallets/pips/src/lib.rs

```diff
@@ -1594,3 +1596,60 @@ mod test {
         assert_eq!(queue, vec![f, d, e, c, a, b, g]);
     }
 }
+
+pub mod migration {
+    use super::*;
+
+    mod v2 {
+        use super::*;
+        use scale_info::TypeInfo;
+
+        /// Represents a proposal
+        #[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq)]
+        #[cfg_attr(feature = "std", derive(Debug))]
+        pub struct Pip<Proposal, AccountId> {
+            /// The proposal's unique id.
+            pub id: PipId,
+            /// The proposal being voted on.
+            pub proposal: Proposal,
+            /// The latest state
+            pub state: ProposalState,
+            /// The issuer of `propose`.
+            pub proposer: Proposer<AccountId>,
+        }
+
+        decl_storage! {
+            trait Store for Module<T: Config> as Pips {
+                /// Actual proposal for a given id, if it's current.
+            /// proposal id -> proposal
+            pub Proposals get(fn proposals): map hasher(twox_64_concat) PipId => Option<Pip<T::Proposal, T::AccountId>>;
+            }
+        }
+
+        decl_module! {
+            pub struct Module<T: Config> for enum Call where origin: T::Origin { }
+        }
+    }
+
+    pub fn migrate_v2<T: Config>() {
+        sp_runtime::runtime_logger::RuntimeLogger::init();
+
+        log::info!(" >>> Updating Pips storage. Migrating Pips...");
+        let total_pips = v2::Proposals::<T>::drain().fold(0usize, |total_pips, (pip_id, pip)| {
+            // Migrate Pips
+            <Proposals<T>>::insert(
+                pip_id,
+                Pip {
+                    id: pip_id,
+                    proposal: pip.proposal,
+                    proposer: pip.proposer.clone(),
```

Comment:
The `clone()` shouldn't be needed here, since we can move out values from the old `pip` value.